ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogCache.java

```diff
@@ -342,10 +342,10 @@ int getMaxCachedSegments() {
   }
 
   void loadSegment(LogPathAndIndex pi, boolean keepEntryInCache,
-      Consumer<LogEntryProto> logConsumer) throws IOException {
+      Consumer<LogEntryProto> logConsumer, long lastIndexInSnapshot) throws IOException {
     LogSegment logSegment = LogSegment.loadSegment(storage, pi.getPath().toFile(),
```

Comment:
We should not load logSegment if endIndex < lastIndexInSnapshot. We will also need to cover the case where endIndex is INVALID_LOG_INDEX.